pinot-core/src/main/java/org/apache/pinot/core/common/RowBasedBlockValueFetcher.java

```diff
@@ -63,12 +65,16 @@ private ValueFetcher createFetcher(BlockValSet blockValSet) {
         case STRING:
           return new StringSingleValueFetcher(blockValSet.getStringValuesSV());
         case BYTES:
+          if (dataType == DataType.BIG_DECIMAL) {
+            return new BigDecimalValueFetcher(blockValSet.getBytesValuesSV());
```

Comment:
Since we added dictionary for big-decimal, we should add the big-decimal to the data access layer (add `BlockValSet.getBigDecimalValuesSV()`)